### PR TITLE
Add Contact DLC JointRails

### PR DESCRIPTION
**Pull request by dedmen**
Is not tested yet as I don't have Contact installed.
Especially muzzle_tma_65 needs testing with all current 6.5 weapons.

We have a incoming problem in that there will be contact-DLC exclusive attachments which will only be included in the optional Contact download. We might need a seperate compat mod for that.

### CODE REVIEW

**Comment by commy2** [comment]
Those classnames are from a discord post. I'd rather wait for the actual config to be out to copy them from there...

Is there a branch to get Contact dev?

**Comment by dedmen** [comment]
>Is there a branch to get Contact dev?

It's on RC and dev. Not sure if both are "current" tho.

**Comment by PabstMirror** [comment]
last I checked there were several JR UBC errors with 1.94 I assumed we would have to fix

**Comment by PabstMirror** [comment]
```
Updating base class MuzzleSlot_65->asdg_MuzzleSlot_762MG, by x\cba\addons\jr\config.cpp/CfgWeapons/LMG_Mk200_F/WeaponSlotsInfo/MuzzleSlot/ (original a3\weapons_f_enoch\machineguns\m200\config.bin)
Updating base class compatibleItems->compatibleItems, by x\cba\addons\jr\config.cpp/CfgWeapons/LMG_Mk200_F/WeaponSlotsInfo/MuzzleSlot/compatibleItems/ (original (no unload))
Updating base class MuzzleSlot_65->asdg_MuzzleSlot_65, by x\cba\addons\jr\config.cpp/CfgWeapons/arifle_MXC_F/WeaponSlotsInfo/MuzzleSlot/ (original (no unload))
Updating base class PointerSlot_Rail->asdg_FrontSideRail, by x\cba\addons\jr\config.cpp/CfgWeapons/arifle_MXC_F/WeaponSlotsInfo/PointerSlot/ (original (no unload))
Updating base class MuzzleSlot_65->asdg_MuzzleSlot_65, by x\cba\addons\jr\config.cpp/CfgWeapons/arifle_MX_F/WeaponSlotsInfo/MuzzleSlot/ (original (no unload))
Updating base class PointerSlot_Rail->asdg_FrontSideRail, by x\cba\addons\jr\config.cpp/CfgWeapons/arifle_MX_F/WeaponSlotsInfo/PointerSlot/ (original (no unload))
Updating base class MuzzleSlot_65->asdg_MuzzleSlot_65, by x\cba\addons\jr\config.cpp/CfgWeapons/arifle_MX_GL_F/WeaponSlotsInfo/MuzzleSlot/ (original (no unload))
Updating base class PointerSlot_Rail->asdg_FrontSideRail, by x\cba\addons\jr\config.cpp/CfgWeapons/arifle_MX_GL_F/WeaponSlotsInfo/PointerSlot/ (original (no unload))
Updating base class MuzzleSlot_65->asdg_MuzzleSlot_65, by x\cba\addons\jr\config.cpp/CfgWeapons/arifle_MX_SW_F/WeaponSlotsInfo/MuzzleSlot/ (original (no unload))
Updating base class PointerSlot_Rail->asdg_FrontSideRail, by x\cba\addons\jr\config.cpp/CfgWeapons/arifle_MX_SW_F/WeaponSlotsInfo/PointerSlot/ (original (no unload))
Updating base class MuzzleSlot_65->asdg_MuzzleSlot_65, by x\cba\addons\jr\config.cpp/CfgWeapons/arifle_MXM_F/WeaponSlotsInfo/MuzzleSlot/ (original (no unload))
Updating base class PointerSlot_Rail->asdg_FrontSideRail, by x\cba\addons\jr\config.cpp/CfgWeapons/arifle_MXM_F/WeaponSlotsInfo/PointerSlot/ (original (no unload))
Updating base class MuzzleSlot_762->asdg_MuzzleSlot_762R, by x\cba\addons\jr\config.cpp/CfgWeapons/arifle_AK12_base_F/WeaponSlotsInfo/MuzzleSlot/ (original (no unload))
Updating base class CowsSlot_Rail->asdg_OpticRail1913, by x\cba\addons\jr\config.cpp/CfgWeapons/arifle_AK12_base_F/WeaponSlotsInfo/CowsSlot/ (original (no unload))
Updating base class UnderBarrelSlot_rail->asdg_UnderSlot, by x\cba\addons\jr\config.cpp/CfgWeapons/arifle_AK12_F/WeaponSlotsInfo/UnderBarrelSlot/ (original (no unload))
Updating base class CowsSlot->asdg_PistolOpticMount, by x\cba\addons\jr\config.cpp/CfgWeapons/hgun_Pistol_heavy_01_F/WeaponSlotsInfo/CowsSlot/ (original a3\weapons_f_enoch\config.bin)
```
are these a problem? or just something we need to do to use CBA JR instead of vanilla?

**Comment by dedmen** [comment]
>are these a problem

I think so... Definitely not nice. Need to double check if config inheritance stuff is correct.

more attachments, exclusive to contact (singleplayer, optional part)
```
optic_Hamr_broken
optic_MRCO_broken
optic_Aco_broken
acc_flashlight_broken
acc_pointer_IR_broken
```

They require contact (weapons_f_contact) which not everyone has. And if we don't add them, we might break the contact campaign.
We remove the vanilla attachments and override it with JR right? so we NEED to add them?

**Comment by commy2** [comment]
We don't remove any vanilla attachments.

**Comment by commy2** [comment]
Contact:
```sqf
weapons[] = {"AlienDrone_01_Sounds_F","arifle_MSBS65_Black_FL_EMP_f","arifle_MSBS65_black_ico_FL_EMP_f","arifle_MSBS65_black_ico_pointer_EMP_f","arifle_MSBS65_GL_black_ico_FL_EMP_f","arifle_MSBS65_GL_black_ico_pointer_EMP_f","arifle_MSBS65_GL_ico_FL_EMP_f","arifle_MSBS65_GL_ico_pointer_EMP_f","arifle_MSBS65_ico_FL_EMP_f","arifle_MSBS65_ico_pointer_EMP_f","arifle_MSBS65_Mark_SOS_LP_BI_EMP_F","arifle_MSBS65_UBS_black_ico_FL_EMP_f","arifle_MSBS65_UBS_black_ico_pointer_EMP_f","arifle_MSBS65_UBS_ico_FL_EMP_f","arifle_MSBS65_UBS_ico_pointer_EMP_f","arifle_MX_Black_FL_EMP_F","arifle_MX_Black_Pointer_EMP_F","arifle_MX_GL_Black_RCO_Pointer_EMP_F","arifle_MX_SW_Black_Pointer_EMP_F","arifle_MXM_Black_MOS_Pointer_Bipod_EMP_F","CM_Universal_01_F","CM_Universal_01_Gravityburst_F","CM_Universal_01_Grenade_F","CM_Universal_01_Hardkill_F","CM_Universal_01_Light_F","GravityCannon_01","GravityCannon_01_Precise","GravityShotgun_01","hgun_SpectrumAnalyzer_F","launch_MRAWS_green_broken_F","LMG_Mk200_black_FL_EMP_F","NVGoggles_INDEP_broken","Rangefinder_broken","SwarmMissile_01_launcher_F"};
weapons[] = {"acc_esd_01_flashlight","acc_flashlight_broken","acc_pointer_IR_broken","muzzleflash_tma","optic_Aco_broken","optic_Hamr_broken","optic_MRCO_broken"};
weapons[] = {"ChemicalDetector_01_base_F","ChemicalDetector_01_black_F","ChemicalDetector_01_olive_F","ChemicalDetector_01_tan_F","ChemicalDetector_01_watch_broken_F","ChemicalDetector_01_watch_F"};
weapons[] = {"hgun_ESD_01_F","hgun_esd_01_antenna_01_F","hgun_esd_01_antenna_02_F","hgun_esd_01_antenna_03_F"};
```

**Comment by commy2** [comment]
Enoch (poland map)
```cpp
weapons[] = {"arifle_MX_Black_ACO_FL_F","arifle_MX_Black_ACO_Pointer_F","arifle_MX_Black_ACO_Pointer_Snds_F","arifle_MX_Black_Hamr_Pointer_Snds_F","arifle_MX_Black_Holo_Pointer_F","arifle_MX_Black_Pointer_F","arifle_MX_GL_Black_ACO_F","arifle_MX_GL_Black_Holo_Pointer_Snds_F","arifle_MX_SW_Black_Pointer_F","arifle_MXC_Black_ACO_F","arifle_MXC_Black_ACO_Pointer_Snds_F","arifle_MXC_Black_Holo_FL_F","arifle_MXC_Black_Holo_Pointer_F","arifle_MXM_Black_MOS_Pointer_Bipod_F","arifle_MXM_Black_MOS_Pointer_Bipod_Snds_F","DeminingDisruptor_01_F","Item_AntidoteKit_01_F","Item_DeconKit_01_F","O_NVGoggles_grn_F","ProbingLaser_01_F","srifle_DMR_04_DMS_weathered_Kir_F_F"};
weapons[] = {"bipod_02_F_arid","bipod_02_F_lush","muzzle_snds_B_arid_F","muzzle_snds_B_lush_F","muzzle_tma_65","optic_Arco_AK_arid_F","optic_Arco_AK_blk_F","optic_Arco_AK_lush_F","optic_Arco_arid_F","optic_Arco_lush_F","optic_DMS_weathered_F","optic_DMS_weathered_Kir_F","optic_Holosight_arid_F","optic_Holosight_lush_F","optic_ico_01_black_f","optic_ico_01_camo_f","optic_ico_01_f","optic_ico_01_sand_f","optic_MRD_black"};
weapons[] = {"I_E_UavTerminal"};
weapons[] = {"launch_RPG32_camo_F","launch_RPG32_green_F"};
weapons[] = {"launch_B_Titan_olive_F","launch_I_Titan_eaf_F"};
weapons[] = {"srifle_DMR_06_hunter_F","srifle_DMR_06_hunter_khs_F"};
weapons[] = {"LMG_Mk200_black_ACO_pointer_F","LMG_Mk200_black_F","LMG_Mk200_black_LP_BI_F"};
weapons[] = {"hgun_Pistol_heavy_01_green_F","hgun_Pistol_heavy_01_green_snds_F"};
weapons[] = {"arifle_AK12_arid_F","arifle_AK12_GL_arid_F","arifle_AK12_GL_lush_arco_pointer_F","arifle_AK12_GL_lush_arco_snds_pointer_F","arifle_AK12_GL_lush_F","arifle_AK12_GL_lush_holo_pointer_F","arifle_AK12_GL_lush_holo_snds_pointer_F","arifle_AK12_lush_arco_pointer_F","arifle_AK12_lush_arco_snds_pointer_bipod_F","arifle_AK12_lush_arco_snds_pointer_F","arifle_AK12_lush_F","arifle_AK12_lush_snds_pointer_F","arifle_AK12U_arid_F","arifle_AK12U_F","arifle_AK12U_lush_F","arifle_AK12U_lush_holo_F","arifle_AK12U_lush_holo_pointer_F","arifle_AK12U_lush_holo_snds_pointer_F","arifle_AK12U_lush_snds_pointer_F","arifle_RPK12_arid_F","arifle_RPK12_F","arifle_RPK12_lush_arco_pointer_F","arifle_RPK12_lush_arco_snds_pointer_F","arifle_RPK12_lush_F","arifle_RPK12_lush_holo_snds_pointer_F"};
weapons[] = {"sgun_HunterShotgun_01_F","sgun_HunterShotgun_01_sawedoff_F"};
weapons[] = {"arifle_MSBS65_black_F","arifle_MSBS65_black_ico_F","arifle_MSBS65_black_ico_pointer_f","arifle_MSBS65_camo_F","arifle_MSBS65_F","arifle_MSBS65_GL_black_F","arifle_MSBS65_GL_black_ico_F","arifle_MSBS65_GL_black_ico_pointer_f","arifle_MSBS65_GL_camo_F","arifle_MSBS65_GL_F","arifle_MSBS65_GL_ico_F","arifle_MSBS65_GL_ico_pointer_f","arifle_MSBS65_GL_sand_F","arifle_MSBS65_ico_F","arifle_MSBS65_ico_pointer_f","arifle_MSBS65_Mark_black_F","arifle_MSBS65_Mark_camo_F","arifle_MSBS65_Mark_F","arifle_MSBS65_Mark_sand_F","arifle_MSBS65_Mark_SOS_LP_BI_F","arifle_MSBS65_sand_F","arifle_MSBS65_UBS_black_F","arifle_MSBS65_UBS_black_ico_F","arifle_MSBS65_UBS_black_ico_pointer_f","arifle_MSBS65_UBS_camo_F","arifle_MSBS65_UBS_F","arifle_MSBS65_UBS_ico_F","arifle_MSBS65_UBS_ico_pointer_f","arifle_MSBS65_UBS_sand_F","arifle_MSBS65_UBS_snds_ico_pointer_F"};
weapons[] = {};
```

**Comment by dedmen** [comment]
so JR only adds things, not remove anything?

**Comment by commy2** [comment]
```

23:10:46 Cannot delete class PointerSlot, it is referenced somewhere (used as a base class probably).

23:10:46 Updating base class MuzzleSlot_65->asdg_MuzzleSlot_65, by x\cba\addons\jr\config.cpp/CfgWeapons/arifle_MX_F/WeaponSlotsInfo/MuzzleSlot/ (original (no unload))
23:10:46 Updating base class PointerSlot_Rail->asdg_FrontSideRail, by x\cba\addons\jr\config.cpp/CfgWeapons/arifle_MX_F/WeaponSlotsInfo/PointerSlot/ (original (no unload))
23:10:46 Updating base class MuzzleSlot_65->asdg_MuzzleSlot_65, by x\cba\addons\jr\config.cpp/CfgWeapons/arifle_MX_GL_F/WeaponSlotsInfo/MuzzleSlot/ (original (no unload))
23:10:46 Updating base class PointerSlot_Rail->asdg_FrontSideRail, by x\cba\addons\jr\config.cpp/CfgWeapons/arifle_MX_GL_F/WeaponSlotsInfo/PointerSlot/ (original (no unload))
23:10:46 Updating base class PointerSlot_Rail->asdg_FrontSideRail, by x\cba\addons\jr\config.cpp/CfgWeapons/arifle_MX_SW_F/WeaponSlotsInfo/PointerSlot/ (original (no unload))
23:10:46 Updating base class PointerSlot_Rail->asdg_FrontSideRail, by x\cba\addons\jr\config.cpp/CfgWeapons/arifle_MXM_F/WeaponSlotsInfo/PointerSlot/ (original (no unload))
23:10:46 Updating base class PointerSlot->asdg_FrontSideRail, by x\cba\addons\jr\config.cpp/CfgWeapons/arifle_AK12_base_F/WeaponSlotsInfo/PointerSlot/ (original a3\weapons_f_exp\rifles\ak12\config.bin)
23:10:46 Updating base class UnderBarrelSlot_rail->asdg_UnderSlot, by x\cba\addons\jr\config.cpp/CfgWeapons/arifle_AK12_F/WeaponSlotsInfo/UnderBarrelSlot/ (original (no unload))
```

**Comment by commy2** [comment]
>so JR only adds things, not remove anything?

Name anything.

**Comment by commy2** [comment]
Just adding classes from optional components like Contact to the compatibleAddons class results in this when the item class does not exist in master config:
```
`No entry 'bin\config.bin/CfgWeapons.optic_Aco_broken'.
```
but I think it can be fixed by more dummy classes :/

**Comment by commy2** [comment]
´"optic_ico_01_black_f","optic_ico_01_camo_f","optic_ico_01_f","optic_ico_01_sand_f"` can be ignored. They are optics that are only supposed to fit on the Promet as far as I can tell.

**Comment by commy2** [comment]
`"acc_esd_01_flashlight","muzzleflash_tma"` can be ignored too. Esd is a flashlight for some alien pistol and tma is already listed in the new vanilla "jr" class for 6.5 (only used by MX).

**Comment by commy2** [comment]
Done. Merge?

File: addons/jr/Cfgweapons.hpp [hunk 1]
```diff
@@ -971,4 +971,30 @@ class CfgWeapons {
             };
         };
     };
+
+    // Contact DLC dummies
+    class acc_flashlight;
+    class acc_flashlight_broken: acc_flashlight {
+        scope = 1; // Optional protected class inheriting from a public base class.
```
Comment (dedmen) [inline addons/jr/Cfgweapons.hpp]: aren't we hiding the vanilla stuff? or is it already hidden.

Also you renamed file from cfgweapons -> Cfgweapons. But shouldn't it be CfgWeapons?

Comment (commy2) [reply addons/jr/Cfgweapons.hpp]: >aren't we hiding the vanilla stuff? or is it already hidden.

These classes are all private in vanilla. But if I inherit when they don't exist, they would get public from their base classes.

>Also you renamed file from cfgweapons -> Cfgweapons. But shouldn't it be CfgWeapons?

Fixed on master

**Pull request merged**